dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/services/SymbolSampleAnalysisEnvironment.kt

```diff
@@ -116,31 +134,102 @@ private class SymbolSampleAnalysisEnvironment(
         }
     }
 
-    private fun processImports(psiElement: PsiElement): List<String> {
+    private fun processImports(psiElement: PsiElement, sampleRewriter: SampleRewriter?): List<String> {
         val psiFile = psiElement.containingFile
         val importsList = (psiFile as? KtFile)?.importList ?: return emptyList()
         return importsList.imports
             .map { it.text.removePrefix("import ") }
             .filter { it.isNotBlank() }
+            .applyIf(sampleRewriter != null) {
+                mapNotNull { sampleRewriter?.rewriteImportDirective(it) }
+            }
     }
 
-    private fun processBody(sampleElement: PsiElement): String {
+    private fun processBody(sampleElement: KtDeclarationWithBody): String {
         return getSampleBody(sampleElement)
             .trim { it == '\n' || it == '\r' }
             .trimEnd()
             .trimIndent()
     }
 
-    private fun getSampleBody(sampleElement: PsiElement): String {
-        return when (sampleElement) {
-            is KtDeclarationWithBody -> {
-                when (val bodyExpression = sampleElement.bodyExpression) {
-                    is KtBlockExpression -> bodyExpression.text.removeSurrounding("{", "}")
-                    else -> bodyExpression!!.text
-                }
-            }
+    private fun getSampleBody(psiElement: KtDeclarationWithBody): String {
```

Comment:
We have discussed it before the big refactoring. The point was  ''it will only complicate things more" since  `analysis-kotlin-descriptors` will be thrown away and only complicated logic will be left.